File: BreaktheChain/java/src/tests/UnitTests.java
package tests;

import static org.junit.Assert.*;

import org.junit.Test;

import breakthechain.Solution;

public class UnitTests {

	@Test
	public void testSolution_ReturnsTheLowestCostToBreakTheChainIntoThreeParts() {
		int[] chain = {5,2,4,6,3,7};
		
        int result = Solution.solution(chain);

        assertEquals(5, result);
	}

}

File: BreaktheChain/csharp/UnitTests/UnitTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using BreaktheChain;

namespace UnitTests
{
    [TestClass]
    public class UnitTests
    {
        [TestMethod]
        public void Test_Solution_ReturnsTheCheapestCostToCutTheChainIntoThreeParts()
        {
            var chain = new[] {5, 2, 4, 6, 3, 7};

            var c = new Solution();
            int result = c.solution(chain);

            Assert.AreEqual(5, result);
        }
    }
}
